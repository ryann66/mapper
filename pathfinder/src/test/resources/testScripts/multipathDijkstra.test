# Test multiple paths existing

CreateGraph mg

AddNode mg A
AddNode mg B
AddNode mg x1
AddNode mg x2
AddNode mg y1
AddNode mg y2

AddEdge mg A x1 7.0
AddEdge mg x1 x2 7.0
AddEdge mg x2 B 7.0
AddEdge mg A y1 8.0
AddEdge mg y1 y2 8.0
AddEdge mg y2 B 8.0

FindPath mg A B

# Test asymmetric shorter path

AddNode mg z1
AddNode mg z2
AddEdge mg A z1 9.0
AddEdge mg z1 z2 9.0
AddEdge mg z2 B 2.0

FindPath mg A B