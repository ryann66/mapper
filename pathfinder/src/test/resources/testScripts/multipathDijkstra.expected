# Test multiple paths existing

created multigraph mg

added node A to mg
added node B to mg
added node x1 to mg
added node x2 to mg
added node y1 to mg
added node y2 to mg

added edge 7.000 from A to x1 in mg
added edge 7.000 from x1 to x2 in mg
added edge 7.000 from x2 to B in mg
added edge 8.000 from A to y1 in mg
added edge 8.000 from y1 to y2 in mg
added edge 8.000 from y2 to B in mg

path from A to B:
A to x1 with weight 7.000
x1 to x2 with weight 7.000
x2 to B with weight 7.000
total cost: 21.000

# Test asymmetric shorter path

added node z1 to mg
added node z2 to mg
added edge 9.000 from A to z1 in mg
added edge 9.000 from z1 to z2 in mg
added edge 2.000 from z2 to B in mg

path from A to B:
A to z1 with weight 9.000
z1 to z2 with weight 9.000
z2 to B with weight 2.000
total cost: 20.000